### PR TITLE
Add test coverage for lib/app/worker.rb + modify heartbeat logic for better separation of concerns

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
This PR adds 100% test coverage for `lib/app/worker.rb` file.

- Additionally, the tests that were checking the warning handling were moved to `spec/utility/es_client_spec.rb` - they seem to belong there better.
- Moved the heartbeat timer start logic into `lib/core/heartbeat.rb` + added tests to it as well

### CODE REVIEW

File: lib/core/heartbeat.rb [hunk 1]
```diff
@@ -6,14 +6,32 @@
...
 require 'connectors/connector_status'
 require 'connectors/registry'
+require 'core/elastic_connector_actions'
 require 'utility/logger'
 
 module Core
   class Heartbeat
     class << self
+      def start_task(connector_id, service_type)
+        interval_seconds = 60 # seconds
```
Comment (tarekziade) [inline lib/core/heartbeat.rb:19]: let's use a CONSTANT

**Review by tarekziade — APPROVED** [review]
just one fly-by neat

YAY TESTS 👯

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/connectors-ruby/pull/245)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/connectors-ruby/pull/247)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by wangch079** [comment]
My backport PR https://github.com/elastic/connectors-ruby/pull/244 is failing and it depends on the merge of backporting of this PR.

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/connectors-ruby/pull/258)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by artem-shelkovnikov** [comment]
@wangch079 I've merged a backport for this PR + the #244 that you've mentioned!

**Pull request merged**